modules/standard/Data/Array.enc

```diff
@@ -85,8 +85,10 @@ end
 -- elements of arr
 fun map[a,b](f : a -> b, arr : [a]) : [b]
   val ret = new [b](|arr|)
-  for i <- [0 .. |arr|-1] do
```

Comment:
I would rather avoid using for-comprehension inside of map, flatMap and foreach, due to the potential endless compilation loop it could result in. Unless I used it with a range or some other kind of data strucutre. But even then, for performance sake, I would rather use a while, or repeat loop inside of these methods.